shub/image/list.py

```diff
@@ -117,9 +117,15 @@ def _run_cmd_in_docker_container(image_name, command, environment):
         raise ShubException("Create container error:\n %s" % container)
     try:
         client.start(container)
-    except docker.errors.NotFound:
-        # return a proper exit code if executable is not found
-        return 127, None
+    except docker.errors.APIError as e:
+        explanation = e.explanation or ''
+        if 'executable file not found' in explanation:
```

Comment:
@vshlapakov tbh I don't think 404 is ever raised in that case. I've seen only 500.